debugging-queries.mdx

```diff
@@ -168,3 +177,128 @@ def sum_transaction_amount(txns: User.transaction[Transaction.amount]) -> User.s
 You can also view these logs using `resolver_replay`, where they will be emitted to your terminal:
 
 ![Logs output example](/img/debugging/logs.png)
+
+## Query Error Categories
+
+In using the debugging techniques above, you may encounter a few different categories of errors
+that have common root causes:
+
+### Request error
+
+Request errors are raised before execution of your resolver code.
+They may occur due to invalid feature names in the
+input or a request that cannot be satisfied by the resolvers you
+have defined.
+
+
+### Field error
+
+Field errors are raised while running a feature resolver
+for a particular field.
+For this type of error, you'll find a `feature` and `resolver`
+attribute in the [error type](#chalkerror).
+
+When a feature resolver crashes, you will receive `null` value
+in the response.
+To differentiate from a resolver returning a `null` value and
+a failure in the resolver, you need to check the error schema.
+
+### Network error
+
+Network errors are thrown outside your resolvers.
+For example, your request was unauthenticated,
+connection failed, or an error occurred within Chalk.
```

Comment:
```suggestion
connection failures, or errors occurring within Chalk.
```
Should we remove errors occurring within Chalk haha or find some way to say it more delicately haha :)

Reply:
i will ~ refine ~